### PR TITLE
Travis needs travis_after_all

### PR DESCRIPTION
**Pull request by honzajavorek**
See

- https://github.com/travis-ci/travis-ci/issues/929
- https://github.com/travis-ci/travis-ci/issues/1548

for details on why this is needed 😫

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -12,20 +12,17 @@ cache:
...
 before_script:
   - "npm run lint"
 script:
   - "npm test"
-after_success:
+after_success:  # travis_after_all.py is needed due to travis-ci/travis-ci#1548 & travis-ci/travis-ci#929
   - "npm run coveralls"
+  - "python travis_after_all.py"
+  - "export $(cat .to_export_back)"
+  - "npm run semantic-release || true"
```
Comment (kylef) [inline .travis.yml]: Won't this automatically release non-master branches too? such as PRs from anyone?

Or does semantic-release have a check later?

Comment (JackuB) [inline .travis.yml]: Semrel is using https://github.com/semantic-release/condition-travis to check whether it is running on master

**Comment by honzajavorek** [comment]
Well this still doesn't work correctly. It just happened to me that new version was published before all jobs of the master build passed 😬

**Comment by honzajavorek** [comment]
Correcting myself after some investigation. When PR is merged into master, Travis CI starts a build. The build releases new version and creates tag. The tag triggers a new Travis CI build, which gets paired with the master branch. For this reason, if you go to GitHub branches and click the "semaphore bullet" for the latest build, you get to the build for tag. This tag build obviously won't finish before the release goes public.

On the other hand previous build, the one for the merge, seems to correctly identify "leader" build job and executes semantic releasing only in that one. All other ones finish with

> This test run is not the build leader and therefore a new version won’t be published.

**Pull request merged**